woocommerce/woocommerce-gutenberg-products-block
# `registerInnerBlock` ->`registerBlockComponent`

This PR repurposes `registerInnerBlock` (which was unused and not fully fit for purpose) into a new registration function named `registerBlockComponent`. For 2 reasons:

- Atomic blocks (based on recent conversations) may not be inner blocks.
- Atomic blocks may not have parents. `registerBlockComponent` parent is optional.

Additionally, `registerInnerBlock` wasn't actually registering blocks. It was registering components to map the blocks. I feel the new naming is clearer.

As well as updating the function and tests, this makes our Atomic blocks register themselves using `registerBlockComponent` in a components-init file.

### How to test the changes in this Pull Request:

1. After building, check for regressions in All Products and Single Products block.
2. Check components still render on the frontend. If so, registration is working.
